lld/test/ELF/linkerscript/discard-symbol.s

```diff
@@ -0,0 +1,16 @@
+# REQUIRES: x86
```

Comment:
The test looks only to check global symbols. The patch makes modifications for local symbols and symbol ordering files, please can you add test cases for these as well?

Will be worth adding a test case for relocatable links too.

Reply:
This is true. The reason I couldn't add a test for the local symbol case is because when you have local symbols, llvm-mc seems to emit them as section symbols with an offset addend rather than an actual symbol relocation. So it runs afoul of the exception for section symbols, and doesn't trigger.

Reply:
Remove the file after Rebase on top of https://github.com/llvm/llvm-project/commit/557299c9b6464f27968904aad5429cfb1512434e